### PR TITLE
add napari to dependencies

### PR DESCRIPTION
**Pull request by haesleinhuepf**
Hi all,

I'm adding `napari` again and again to this file in napari-plugins I make. I'm suggesting to make `napari` a dependency of all napari-plugins generated with the cookie-cutter. Is there a scenario where this dependency makes no sense?

Best,
Robert

### CODE REVIEW

**Comment by tlambert03** [comment]
this goes back to some early discussions about plugin design.  Certainly, there are plenty of cases where a plugin will want to directly use the napari API.  But it was also important to us that something could offer up functionality as a napari plugin without _requiring_ that the entire napari dependency stack be installed in the environment.  For instance, think about a reader plugin that knows how to take a path and return data... none of that requires Qt, or vispy, or all of the heavy dependencies that come with napari.  One could definitely imagine wanting to reuse that code in a non-napari context (see https://github.com/napari/napari-plugin-engine/issues/24 for more).
So while I think that the majority of your napari plugins have been designed _only_ with napari in mind, that's not always the case (among many examples, see https://github.com/tlambert03/ome-types ... where I _definitely_ don't want to depend on napari, but if `ome-types` happens to be installed in a napari environment, i'd like napari to be able to use it).

That said, I don't feel too strongly either way here.  The cookiecutter is a little bit of a special case... and it's just as easy to _remove_ napari from the dependencies as it is to add it.  For the reasons mentioned above, I think I lean slightly against adding it to the dependencies, but wouldn't be totally opposed if @jni or @sofroniewn  wanted it in there

**Comment by tlambert03** [comment]
you can kind of check out the "landscape" with the following query on google big query:

```sql
SELECT  name, home_page, requires_dist
    FROM `bigquery-public-data.pypi.distribution_metadata`
    LEFT JOIN UNNEST(classifiers) cl
    WHERE cl='Framework :: napari'
```

https://console.cloud.google.com/bigquery?sq=393501463861:8ccdf2ee913849e1805136289e3c531f

you can see how many plugins use napari.   it's a bit hard to parse, since that's unique distributions (versions) rather than unique packages.  But a rough estimation suggests that "most" distributions haven't added napari to their dependencies:
 
Untitled

**Comment by haesleinhuepf** [comment]
Wow this query is amazing! 🤩 

Can you also parse for plugins that import napari but don't have napari as dependency?
Example [dependencies](https://github.com/haesleinhuepf/napari-manual-split-and-merge-labels/blob/main/setup.cfg#L34) and [import](https://github.com/haesleinhuepf/napari-manual-split-and-merge-labels/blob/main/napari_manual_split_and_merge_labels/_function.py#L6)

If I'm the only one who does these things wrong, I'm fine with not merging this. You know your target-audience better. 🙂

**Comment by tlambert03** [comment]
> Can you also parse for plugins that import napari but don't have napari as dependency?

the pypi database on big query is just the package metadata (what you put in your setup.cfg or setup.py), so it doesn't have any source code (one could grab the github urls from the metadata and parse though).

In any case, you probably shouldn't import napari without either declaring it as a dependency 😁 ... so i'd add `napari` to deps again in that case.

> If I'm the only one who does these things wrong, I'm fine with not merging this. You know your target-audience better. 🙂

I wouldn't say you're doing anything "wrong".  There's absolutely nothing wrong with using the napari API from within napari, and depending on it.  _But_ it's not required..., and I think we'll probably stick with that pattern where it makes sense (like for reader plugins) ...
it's in the interest of fostering a "lean" ecosystem of packages with minimal heavy inter-dependencies that we avoided encouraging that type of usage.
